TELCODOCS-28: New status items, drainProgress and elapsedTime, added

ogradyp opened the pull request:
https://issues.redhat.com/browse/TELCODOCS-28 D/S Doc & RN: ECOPROJECT-2/ [KNIDEPLOY-4010](https://issues.redhat.com//browse/KNIDEPLOY-4010) NMO: Improved admin feedback

Applies to OpenShift version : 4.11 +

Preview:
1. [Checking status of current NodeMaintenance CR tasks](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-checking_status_of_node_maintenance_cr_tasks_node-maintenance-operator)

2. [Setting a node to maintenance mode in the web console using the Actions control](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-actions-web-console_node-maintenance-operator)

3. [Resuming a node from maintenance mode in the web console using the Actions control](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-resuming-node-maintenance-actions-web-console_node-maintenance-operator)


@razo7 can you please review/approve:
- the new status items for [ECOPROJECT-57] (https://issues.redhat.com/browse/ECOPROJECT-57) at 1 above?
- the new module for [ECOPROJECT-536] (https://issues.redhat.com/browse/ECOPROJECT-536) at 2 above?
- the new module for [ECOPROJECT-536] (https://issues.redhat.com/browse/ECOPROJECT-536) at 3 above?

Thank you.

razo7 commented:
It looks great but I still have some comments :)

1.  [Checking status of current NodeMaintenance CR tasks](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-checking_status_of_node_maintenance_cr_tasks_node-maintenance-operator) - Looks wonderful, and I would like to see it as subsubsection after [_Setting a node to maintenance mode in the web console using the Actions control_](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-actions-web-console_node-maintenance-operator) and before [_Resuming a node from maintenance mode_](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#resuming-node-from-maintenance-mode). We didn't have this it as subsubsection in [OCP 4.10](https://docs.openshift.com/container-platform/4.10/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-cr-cli_node-maintenance-operator).
2. [Setting a node to maintenance mode in the web console using the Actions control](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-actions-web-console_node-maintenance-operator) - Looks wonderful
3. I would like to add the _Resuming a node from maintenance mode in the web console_ subsection as it was in [OpenShift Virtualization 4.10](https://docs.openshift.com/container-platform/4.10/virt/node_maintenance/virt-resuming-node.html#virt-resuming-node-maintenance-web_virt-resuming-node) to be _Resuming a node to maintenance mode in the web console using the Actions control_ after [Resuming a node from maintenance mode by using the CLI](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-resuming-node-from-maintenance-mode-with-cr_node-maintenance-operator).
It should have the same first step under _Procedure_ as in [Setting a node to maintenance mode in the web console using the Actions control](https://github.com/openshift/openshift-docs/pull/46914/files#diff-60dd673ef5abd020eb920b44cc709fcf08e31987ef9a1567c5dde229114c30d0R13), and the last paragraph should be as follows.

`The node becomes schedulable. If it had virtual machine instances that were running on the node prior to maintenance, then they will not automatically migrate back to this node.`

ogradyp commented:
Thanks for your feedback @razo7 
I have applied the changes you suggested in points 1 and 3.
Can you please re-review? 
Many Thanks

razo7 commented:
We state `custom resources (CRs)` in five different places alongside one page:

- [about the Node Maintenance Operator](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-about-node-maintenance-operator_node-maintenance-operator).
- [installing the Node Maintenance Operator by using the CLI](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-maintenance-operator-using-cli_node-maintenance-operator) 
- [setting a node to maintenance mode by using the web console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-web-console_node-maintenance-operator)
- [setting a node to maintenance mode by using the CLI](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-cr-cli_node-maintenance-operator)
- [resuming a node from maintenance mode by using the web console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-resuming-node-maintenance-web-console_node-maintenance-operator)

Do you think we need to state the acronym five times or one time at the beginning of the page is enough?

razo7 commented:
Because we are adding the _**using the Actions control**_ option for setting and resuming a node to maintenance mode, then it should be added into the section details in [setting a node to maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#setting-node-in-maintenance-mode) and in [resuming a node from maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#resuming-node-from-maintenance-mode).

For instance in [setting a node to maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#setting-node-in-maintenance-mode) we can write:

> You can place a node into maintenance from the web console or CLI, or even the web console action control (only) for bare-metal cluster, by using a NodeMaintenance CR.

razo7 commented:
And in [resuming a node from maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#resuming-node-from-maintenance-mode):

> You can resume a node from maintenance mode from the web console or CLI, or even the web console action control (only) for bare-metal cluster, by using a NodeMaintenance CR. Resuming a node brings it out of maintenance mode, and makes it schedulable again.

razo7 commented:
I am not sure the information  at the _**If the Operator is not installed successfully**_ under _verification_ part in [installing the Node Maintenance Operator by using the web console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-node-maintenance-operator-using-web-console_node-maintenance-operator) is correct.

1. I think the second step should point to NMO pod and not any pod
2. The second step, _Workloads → Pods_ , can also be an another option to _Operators → Installed Operators_ rather than a sequential two steps verification procedure.

What do you think? @anna-savina

anna-savina commented:
Everything looks good, just few small comments:

1. Since 4.11 we don't need this Note any more
`If you have installed OpenShift Virtualization, you must use the Node Maintenance Operator that is bundled with it.`

2. I suggest to  change "If the Operator is not installed successfully" section in this manner:

```
- Navigate to the Operators → Installed Operators page and inspect the Status column for any errors or failures.
- Navigate to the Operators → Installed Operators -> node-maintenance-operator Details page and inspect errors in "Conditions" sections (could be errors before pod creations)
- Navigate to the Workloads → Pods page and check the logs in node-maintenance-operator-controller-manager pod
```

3. At the end of "Setting a node to maintenance mode in the web console using the Actions control" section could be useful to add

```
Verification
Navigate to the Compute/Nodes and verify that corresponding node has status "Under maintenance"
```

4. At the end of "Resuming a node from maintenance mode in the web console using the Actions control" section also could be useful to add Verification
```
Verification
Navigate to the Compute/Nodes and verify that corresponding node has status Ready
```

anna-savina commented:
Also do we want to describe new power options for Compute/Bare Metal Hosts tab?
Since UI support NMO stand-alone we can perform gracefully actions:
- power on bmh
- power off bmh
- restart bmh

But maybe because these options are too specific and applied just to bmh clusters, we shouldn't mention it.
@razo7 WDYT

ogradyp commented:
> We state `custom resources (CRs)` in five different places alongside one page:
> 
> * [about the Node Maintenance Operator](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-about-node-maintenance-operator_node-maintenance-operator).
> * [installing the Node Maintenance Operator by using the CLI](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-maintenance-operator-using-cli_node-maintenance-operator)
> * [setting a node to maintenance mode by using the web console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-web-console_node-maintenance-operator)
> * [setting a node to maintenance mode by using the CLI](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-cr-cli_node-maintenance-operator)
> * [resuming a node from maintenance mode by using the web console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-resuming-node-maintenance-web-console_node-maintenance-operator)
> 
> Do you think we need to state the acronym five times or one time at the beginning of the page is enough?

Or, what you suggest is correct. However, there are many individual modules within this Operator content and may be referenced as needed, rather than the entire content read from top to bottom. For that reason, and given that it has content has been reviewed before for previous releases, I would leave it as is.

ogradyp commented:
> Because we are adding the _**using the Actions control**_ option for setting and resuming a node to maintenance mode, then it should be added into the section details in [setting a node to maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#setting-node-in-maintenance-mode) and in [resuming a node from maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#resuming-node-from-maintenance-mode).
> 
> For instance in [setting a node to maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#setting-node-in-maintenance-mode) we can write:
> 
> > You can place a node into maintenance from the web console or CLI, or even the web console action control (only) for bare-metal cluster, by using a NodeMaintenance CR.

Or, you are correct to point this out and I have updated the content as follows: [Setting a node to maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#setting-node-in-maintenance-mode)

ogradyp commented:
> And in [resuming a node from maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#resuming-node-from-maintenance-mode):
> 
> > You can resume a node from maintenance mode from the web console or CLI, or even the web console action control (only) for bare-metal cluster, by using a NodeMaintenance CR. Resuming a node brings it out of maintenance mode, and makes it schedulable again.

Or, you are correct to point this out and I have updated the content as follows: [Resuming a node from maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#resuming-node-from-maintenance-mode)

ogradyp commented:
> I am not sure the information at the _**If the Operator is not installed successfully**_ under _verification_ part in [installing the Node Maintenance Operator by using the web console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-node-maintenance-operator-using-web-console_node-maintenance-operator) is correct.
> 
> 1. I think the second step should point to NMO pod and not any pod
> 2. The second step, _Workloads → Pods_ , can also be an another option to _Operators → Installed Operators_ rather than a sequential two steps verification procedure.
> 
> What do you think? @anna-savina

Or, I have addressed this under Anna's comments.

ogradyp commented:
> Everything looks good, just few small comments:
> 
> 1. Since 4.11 we don't need this Note any more
>    `If you have installed OpenShift Virtualization, you must use the Node Maintenance Operator that is bundled with it.`
> 2. I suggest to  change "If the Operator is not installed successfully" section in this manner:
> 
> ```
> - Navigate to the Operators → Installed Operators page and inspect the Status column for any errors or failures.
> - Navigate to the Operators → Installed Operators -> node-maintenance-operator Details page and inspect errors in "Conditions" sections (could be errors before pod creations)
> - Navigate to the Workloads → Pods page and check the logs in node-maintenance-operator-controller-manager pod
> ```
> 
> 3. At the end of "Setting a node to maintenance mode in the web console using the Actions control" section could be useful to add
> 
> ```
> Verification
> Navigate to the Compute/Nodes and verify that corresponding node has status "Under maintenance"
> ```
> 
> 4. At the end of "Resuming a node from maintenance mode in the web console using the Actions control" section also could be useful to add Verification
> 
> ```
> Verification
> Navigate to the Compute/Nodes and verify that corresponding node has status Ready
> ```

@anna-savina, I have addressed your points as follows:
1. [Using the Node Maintenance Operator to place nodes in maintenance mode](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html)
2. [installing the Node Maintenance Operator by using the web console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-node-maintenance-operator-using-web-console_node-maintenance-operator)**
3. [Setting a node to maintenance mode in the web console using the Actions control](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-actions-web-console_node-maintenance-operator)
4. [Resuming a node from maintenance mode in the web console using the Actions control](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-resuming-node-maintenance-actions-web-console_node-maintenance-operator)

ogradyp commented:
> Also do we want to describe new power options for Compute/Bare Metal Hosts tab? Since UI support NMO stand-alone we can perform gracefully actions:
> 
> * power on bmh
> * power off bmh
> * restart bmh
> 
> But maybe because these options are too specific and applied just to bmh clusters, we shouldn't mention it. @razo7 WDYT

@razo7 @anna-savina I have not yet addressed this comment. Or do you want it included as Anna suggests?

anna-savina commented:
@ogradyp except for one comment above, content looks good for me

But I have 2 small recommendations:
1. I don't see `"Checking status of current NodeMaintenance CR tasks" `in the `Table of Contents` on a right side of page. Might be worth adding this section to the Table of Contents
2. Add Verification to all sections:
- In the section` "Setting a node to maintenance mode by using the CLI"` separate 3rd step `"Check the progress of the maintenance task by running the following command" ` and name it _Verification._
- In the section `"Resuming a node from maintenance mode by using the CLI"` add
_Verification:_
run command:
`oc get nodes`
and verify that the corresponding node has a status of Ready

razo7 commented:
> > Also do we want to describe new power options for Compute/Bare Metal Hosts tab? Since UI support NMO stand-alone we can perform gracefully actions:
> > 
> > * power on bmh
> > * power off bmh
> > * restart bmh
> > 
> > But maybe because these options are too specific and applied just to bmh clusters, we shouldn't mention it. @razo7 WDYT
> 
> @razo7 @anna-savina I have not yet addressed this comment. Or do you want it included as Anna suggests?

It was never introduced in OpenShfit Virtualization, where it is only for bare metal clusters, and I doubt we need it. Do you find it useful? I have never been aware of these options.

razo7 commented:
> Everything looks good, just few small comments:
> 
> 1. Since 4.11 we don't need this Note any more
>    `If you have installed OpenShift Virtualization, you must use the Node Maintenance Operator that is bundled with it.`

What do you think about saying `In case OpenShift Virtualization v4.10.z or less is installed in your cluster, then it has an old Node Maintenance Operator version. Please update OpenShift Virtualization version, and use this newer Node Maintenance Operator.` ?
> 2. I suggest to  change "If the Operator is not installed successfully" section in this manner:
> 
> ```
> - Navigate to the Operators → Installed Operators page and inspect the Status column for any errors or failures.
> - Navigate to the Operators → Installed Operators -> node-maintenance-operator Details page and inspect errors in "Conditions" sections (could be errors before pod creations)
> ```

I would change it to -> `Navigate to the Operators → Installed Operators -> Node Maintenance Operator -> Details page and inspect errors in "Conditions" sections (could be errors before pod creations)`

> ```
> - Navigate to the Workloads → Pods page and check the logs in node-maintenance-operator-controller-manager pod
> ```

I would change it to -> `Navigate to the Workloads → Pods page, search for node-maintenance-operator-controller-manager pod in the installed namespace (e,g, openshift-operators), and see the logs under Logs tab`

ogradyp commented on modules/eco-node-maintenance-operator-installation-web-console.adoc:
```diff
@@ -29,4 +29,5 @@ To confirm that the installation is successful:
 
 . Navigate to the *Operators* -> *Installed Operators* page and inspect the `Status` column for any errors or failures.
-. Navigate to the *Workloads* -> *Pods* page and check the logs in any pods in the `openshift-operators` project that are reporting issues.
+. Navigate to the *Operators* -> *Installed Operators* -> *Node-maintenance-operator Details* page and inspect errors in the `Conditions` sections. There may be errors before pods were created.
```
Thanks @anna-savina , updated now

[Installing the Node Maintenance Operator by using the web console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-node-maintenance-operator-using-web-console_node-maintenance-operator)

razo7 replied:
1. @ogradyp  I don't see your change in https://github.com/openshift/openshift-docs/pull/46914/files#diff-d40517f6dbef6eeac9fc5d33b3c43b566f531a4c384a7fb4b1fa8533e1130ca3R32 for [About the Node Maintenance Operator](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-about-node-maintenance-operator_node-maintenance-operator).
2. Also there ate two additional plus signs (+) and one dot sign (.) that I believe we can dismiss, see https://github.com/openshift/openshift-docs/pull/46914/files#diff-617d99b26aae91621ad418181458913442c3d74c707503f4a1bb88ff05d82886R43 for [Setting a node to maintenance mode by using the CLI](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-cr-cli_node-maintenance-operator).
3. How about writing it as one line (pause) and then the command rather than one line (pause), command(pause), and a new line (which is part of the original line..)?
Also, it might be better to use a one method/command for verification that the node is (or not) under maintenance for [Setting a node to maintenance mode by using the CLI](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-cr-cli_node-maintenance-operator) & [Resuming a node from maintenance mode by using the CLI](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-resuming-node-from-maintenance-mode-with-cr_node-maintenance-operator).
`oc describe node <node-name>` sounds very good to me.
@anna-savina WDYT?
If so then I would pretty much copy paste the verification method from [Setting a node to maintenance mode by using the CLI](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-cr-cli_node-maintenance-operator), without the dot and plus signs, and only add a new `Example output`.

ogradyp commented:
> In each page we see `custom resources (CRs)` only once.

@razo7 Updated as per your examples and suggestion. There is now only one occurrence of custom resources (CRs) in the first instance at the top [About the Node Maintenance operator](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-about-node-maintenance-operator_node-maintenance-operator)

ogradyp commented:
> > > Also do we want to describe new power options for Compute/Bare Metal Hosts tab? Since UI support NMO stand-alone we can perform gracefully actions:
> > > 
> > > * power on bmh
> > > * power off bmh
> > > * restart bmh
> > > 
> > > But maybe because these options are too specific and applied just to bmh clusters, we shouldn't mention it. @razo7 WDYT
> > 
> > 
> > @razo7 @anna-savina I have not yet addressed this comment. Or do you want it included as Anna suggests?
> 
> It was never introduced in OpenShfit Virtualization, where it is only for bare metal clusters, and I doubt we need it. Do you find it useful? I have never been aware of these options.

@razo7 I don't think they need documenting.

ogradyp commented:
> > Everything looks good, just few small comments:
> > 
> > 1. Since 4.11 we don't need this Note any more
> >    `If you have installed OpenShift Virtualization, you must use the Node Maintenance Operator that is bundled with it.`
> 
> What do you think about saying `In case OpenShift Virtualization v4.10.z or less is installed in your cluster, then it has an old Node Maintenance Operator version. Please update OpenShift Virtualization version, and use this newer Node Maintenance Operator.` ?
> 
> > 2. I suggest to  change "If the Operator is not installed successfully" section in this manner:
> > 
> > ```
> > - Navigate to the Operators → Installed Operators page and inspect the Status column for any errors or failures.
> > - Navigate to the Operators → Installed Operators -> node-maintenance-operator Details page and inspect errors in "Conditions" sections (could be errors before pod creations)
> > ```
> 
> I would change it to -> `Navigate to the Operators → Installed Operators -> Node Maintenance Operator -> Details page and inspect errors in "Conditions" sections (could be errors before pod creations)`
> 
> > ```
> > - Navigate to the Workloads → Pods page and check the logs in node-maintenance-operator-controller-manager pod
> > ```
> 
> I would change it to -> `Navigate to the Workloads → Pods page, search for node-maintenance-operator-controller-manager pod in the installed namespace (e,g, openshift-operators), and see the logs under Logs tab`

@razo7 

1. I don't think the Note is required so I've removed it as per @anna-savina suggestion
2. I've updated as per @anna-savina suggestion as her initial comment was just for my informaton, which I misunderstood. 
3. Updates as per suggestion, see 

[Installing the Node Maintenance Operator by using the Web Console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-node-maintenance-operator-using-web-console_node-maintenance-operator)

razo7 commented:
> > > Everything looks good, just few small comments:
> > > 
> > > 1. Since 4.11 we don't need this Note any more
> > >    `If you have installed OpenShift Virtualization, you must use the Node Maintenance Operator that is bundled with it.`
> > 
> > 
> > What do you think about saying `In case OpenShift Virtualization v4.10.z or less is installed in your cluster, then it has an old Node Maintenance Operator version. Please update OpenShift Virtualization version, and use this newer Node Maintenance Operator.` ?
> > > 2. I suggest to  change "If the Operator is not installed successfully" section in this manner:
> > > 
> > > ```
> > > - Navigate to the Operators → Installed Operators page and inspect the Status column for any errors or failures.
> > > - Navigate to the Operators → Installed Operators -> node-maintenance-operator Details page and inspect errors in "Conditions" sections (could be errors before pod creations)
> > > ```
> > 
> > 
> > I would change it to -> `Navigate to the Operators → Installed Operators -> Node Maintenance Operator -> Details page and inspect errors in "Conditions" sections (could be errors before pod creations)`
> > > ```
> > > - Navigate to the Workloads → Pods page and check the logs in node-maintenance-operator-controller-manager pod
> > > ```
> > 
> > 
> > I would change it to -> `Navigate to the Workloads → Pods page, search for node-maintenance-operator-controller-manager pod in the installed namespace (e,g, openshift-operators), and see the logs under Logs tab`
> 
> @razo7
> 
> 1. I don't think the Note is required so I've removed it as per @anna-savina suggestion.
> 2. I've updated as per @anna-savina suggestion as her initial comment was just for my informaton, which I misunderstood.
> 3. Updates as per suggestion, see
> 
> [Installing the Node Maintenance Operator by using the Web Console](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-node-maintenance-operator-using-web-console_node-maintenance-operator).

I have noticed your changes, but I still  believe comments/suggestions should be used. 

1. The note that has been removed was more general, I guess for a user with OpenShift Virtualiztion 4.11, but if has new OCP version and older OpenShift Virtualiztion, e.g. 4.10.z and older, then we don't the user to use the old NMO...

2+3. For instance, yesterday I tried to reproduce these steps, and I have noticed that there is `Node-maintenance-operator` it is `Node Maintenance Operator`. 

Do you agree? @anna-savina

anna-savina commented:
1.  > I have noticed your changes, but I still believe comments/suggestions should be used.

yes, you are right. So, probably, we still need that Note, just have to specify version of CNV
`If you have installed OpenShift Virtualization version less then 4.11.0, you must use the Node Maintenance Operator that is bundled with it.`

2. > 2+3. For instance, yesterday I tried to reproduce these steps, and I have noticed that there is Node-maintenance-operator it is Node Maintenance Operator.

yes, you are right. Correct name is Node Maintenance Operator

3. In `Setting a node to maintenance mode by using the CLI` Verification looks a little bit weird 

Do you agree? @razo7

razo7 commented:
> 1. > I have noticed your changes, but I still believe comments/suggestions should be used.
> 
> yes, you are right. So, probably, we still need that Note, just have to specify version of CNV `If you have installed OpenShift Virtualization version less then 4.11.0, you must use the Node Maintenance Operator that is bundled with it.`
> 
> 2. > 2+3. For instance, yesterday I tried to reproduce these steps, and I have noticed that there is Node-maintenance-operator it is Node Maintenance Operator.
> 
> yes, you are right. Correct name is Node Maintenance Operator
> 
> 3. In `Setting a node to maintenance mode by using the CLI` Verification looks a little bit weird
> 
> Do you agree? @razo7



1. I don't. We don't want to suggest users to use old, CNV NMO, on the new NMO page. @ogradyp please use my note.
`In case OpenShift Virtualization v4.10.z or less is installed in your cluster, then it has an old Node Maintenance Operator version. Please update OpenShift Virtualization version, and use this newer Node Maintenance Operator.`

2+3: wonderful.

ogradyp commented:
> > 1. > I have noticed your changes, but I still believe comments/suggestions should be used.
> > 
> > yes, you are right. So, probably, we still need that Note, just have to specify version of CNV `If you have installed OpenShift Virtualization version less then 4.11.0, you must use the Node Maintenance Operator that is bundled with it.`
> > 
> > 2. > 2+3. For instance, yesterday I tried to reproduce these steps, and I have noticed that there is Node-maintenance-operator it is Node Maintenance Operator.
> > 
> > yes, you are right. Correct name is Node Maintenance Operator
> > 
> > 3. In `Setting a node to maintenance mode by using the CLI` Verification looks a little bit weird
> > 
> > Do you agree? @razo7
> 
> 1. I don't. We don't want to suggest users to use old, CNV NMO, on the new NMO page. @ogradyp please use my note.
>    `In case OpenShift Virtualization v4.10.z or less is installed in your cluster, then it has an old Node Maintenance Operator version. Please update OpenShift Virtualization version, and use this newer Node Maintenance Operator.`
> 
> 2+3: wonderful.

@razo7 Updated as per 1, 2 + 3 above. Here is the latest preview with all the changes http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html

johnwilkins commented on applications/connecting_applications_to_services/odc-connecting-an-application-to-a-service-using-the-developer-perspective.adoc:
```diff
@@ -5,93 +5,22 @@ include::_attributes/common-attributes.adoc[]
 In addition to grouping multiple components within an application, you can also use the *Topology* view to connect components with each other. You can either use a binding connector or a visual one to connect components.
 
-A binding connection between the components can be established only if the target node is an Operator-backed service. This is indicated by the *Create a binding connector* tool-tip which appears when you drag an arrow to such a target node. When an application is connected to a service using a binding connector a `ServiceBinding` resource is created. Then, the Service Binding Operator controller  ...
+A binding connection between the components can be established only if the target node is an Operator-backed service. This is indicated by the *Create a binding connector* tool-tip which appears when you drag an arrow to such a target node. When an application is connected to a service by using a binding connector a `ServiceBinding` resource is created. Then, the Service Binding Operator controll ...
```
1. Add a comma before "which"
2. Add a comma after "binding connector" Also, this sentence uses two be verbs, which is passive. If you know the subject of the action, you could make the sentence a bit stronger. In the first part of the sentence, you could say, "When an application uses a binding connector to connect to a service, ..." Then, "<something> creates a ServiceBinding resource. This eliminates the passive voice.

ogradyp replied:
Apologies John, commits from other writers (including this file) got included in my PR. Therefore, I do not include this feedback.

johnwilkins commented on applications/connecting_applications_to_services/odc-connecting-an-application-to-a-service-using-the-developer-perspective.adoc:
```diff
@@ -5,93 +5,22 @@ include::_attributes/common-attributes.adoc[]
-A binding connection between the components can be established only if the target node is an Operator-backed service. This is indicated by the *Create a binding connector* tool-tip which appears when you drag an arrow to such a target node. When an application is connected to a service using a binding connector a `ServiceBinding` resource is created. Then, the Service Binding Operator controller  ...
+A binding connection between the components can be established only if the target node is an Operator-backed service. This is indicated by the *Create a binding connector* tool-tip which appears when you drag an arrow to such a target node. When an application is connected to a service by using a binding connector a `ServiceBinding` resource is created. Then, the Service Binding Operator controll ...
 
 A visual connector establishes only a visual connection between the components, depicting an intent to connect. No interaction between the components is established. If the target node is not an Operator-backed service the *Create a visual connector* tool-tip is displayed when you drag an arrow to a target node.
```
A similar comment regarding passive voice. The second sentence implies the visual connection of the first sentence as the subject. It could be rephrased as, "The visual connection does not establish any interaction between the components." 

In the second sentence, you should put a comma after service, indicating a subordinated clause. "is displayed" is also passive. Maybe, ", the user interface displays the Create a visual connector tool tip..."

ogradyp replied:
Apologies John, commits from other writers (including this file) got included in my PR. Therefore, I do not include this feedback.

johnwilkins commented on modules/odc-creating-a-binding-connection-between-components.adoc:
```diff
@@ -0,0 +1,65 @@
+.Prerequisites
+
+* You have created and deployed a Spring PetClinic sample application by using the *Developer* perspective.
+* You have installed {servicebinding-title} from the OperatorHub.
```
the {servicebinding-title}

ogradyp replied:
Apologies John, commits from other writers (including this file) got included in my PR. Therefore, I do not include this feedback.

johnwilkins commented on nodes/nodes/eco-node-maintenance-operator.adoc:
```diff
@@ -10,7 +10,7 @@ You can use the Node Maintenance Operator to place nodes in maintenance mode. Th
 [NOTE]
 ====
-If you have installed {VirtProductName}, you must use the Node Maintenance Operator that is bundled with it.
+If {VirtProductName} v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version. Upgrade {VirtProductName} to avail of the latest version of the Node Maintenance Operator.
```
s/avail of/obtain/

ogradyp replied:
Thanks John, I've included this feedback.

johnwilkins reviewed (commented):
I had just a few comments. Mostly restructuring sentences with passive voice and ensuring you don't omit articles.

ogradyp commented on modules/cli-installing-cli-rpm.adoc:
```diff
@@ -43,20 +43,16 @@ For {op-system-base-full}, you can install the OpenShift CLI (`oc`) as an RPM if
++
+[NOTE]
+====
+It is not supported to install the OpenShift CLI (`oc`) as an RPM for {op-system-base-full} 9. You must install the OpenShift CLI for {op-system-base} 9 by downloading the binary.
```
Apologies John, commits from other writers (including this file) got included in my PR. Therefore, I do not include this feedback.

ogradyp commented on modules/odc-creating-a-binding-connection-between-components.adoc:
```diff
@@ -0,0 +1,65 @@
+You can establish a binding connection with Operator-backed components.
+
+This procedure walks through an example of creating a binding connection between a PostgreSQL Database service and a Spring PetClinic sample application. To create a binding connection with a service that is backed by the PostgreSQL Database Operator, you must first add the Red Hat-provided PostgreSQL Database Operator to the *OperatorHub*, and then install the Operator.
+The PostreSQL Database Operator then creates and manages the `Database` resource, which exposes the binding information in secrets, config maps, status, and spec attributes.
```
Apologies John, commits from other writers (including this file) got included in my PR. Therefore, I do not include this feedback.

ogradyp commented on modules/odc-creating-a-binding-connection-between-components.adoc:
```diff
@@ -0,0 +1,65 @@
+      resource: deployments
+----
+
+A service binding request is created and the {servicebinding-title} controller projects the database service connection information into the application deployment as files by using a volume mount. After the request is successful, the application is redeployed and the connection is established.
```
Apologies John, commits from other writers (including this file) got included in my PR. Therefore, I do not include this feedback.

ogradyp commented on modules/eco-node-maintenance-operator-installation-web-console.adoc:
```diff
@@ -29,4 +29,5 @@ To confirm that the installation is successful:
 
 . Navigate to the *Operators* -> *Installed Operators* page and inspect the `Status` column for any errors or failures.
-. Navigate to the *Workloads* -> *Pods* page and check the logs in any pods in the `openshift-operators` project that are reporting issues.
+. Navigate to the *Operators* -> *Installed Operators* -> *Node Maintenance Operator* -> *Details* page, and inspect for errors before pod creation, in the `Conditions` sections.
```
Thanks John, I've included this feedback.

ogradyp commented on modules/eco-node-maintenance-operator-installation-web-console.adoc:
```diff
@@ -29,4 +29,5 @@ To confirm that the installation is successful:
 . Navigate to the *Operators* -> *Installed Operators* page and inspect the `Status` column for any errors or failures.
-. Navigate to the *Workloads* -> *Pods* page and check the logs in any pods in the `openshift-operators` project that are reporting issues.
+. Navigate to the *Operators* -> *Installed Operators* -> *Node Maintenance Operator* -> *Details* page, and inspect for errors before pod creation, in the `Conditions` sections.
+. Navigate to the *Workloads* -> *Pods* page, search for `Node Maintenance Operator` pod in the installed namespace, and check the logs in the `Logs` tab.
```
Thanks John, I've included this feedback.

ogradyp commented on modules/eco-resuming-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,30 @@
+
+:_content-type: PROCEDURE
+[id="eco-resuming-node-maintenance-actions-web-console_{context}"]
+
```
Thanks John, I've included this feedback.

ogradyp commented on modules/eco-resuming-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,30 @@
+= Resuming a node from maintenance mode in the web console using the Actions control
+Resume a node from maintenance mode using the Options menu {kebab} found
+on each node in the *Compute* -> *Nodes* list, or using the *Actions* control
+of the *Node Details* screen.
```
Thanks John, I've included this feedback.

ogradyp commented on modules/eco-resuming-node-maintenance-cr-cli.adoc:
```diff
@@ -28,3 +28,22 @@ $ oc delete -f nodemaintenance-cr.yaml
+$ oc describe node node-11.example.com
+----
++
+. Check the example output
```
Thanks John, I've included this feedback.

ogradyp commented on modules/eco-setting-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,35 @@
+
+. From the *Administrator* perspective, in the web console, click *Compute* -> *Nodes*.
+. You can set the node to maintenance from this screen, which makes it easier to perform
+actions on multiple nodes in the one screen or from the *Node Details* screen
```
Thanks John, I've included this feedback.

ogradyp commented on modules/eco-setting-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,35 @@
+. Click *Start Maintenance* in the confirmation window.
+
+The node is no longer schedulable. If it had virtual machines with the `LiveMigration` eviction strategy,
+then it will live migrate them. All other pods and virtual machines, if any, on the node are deleted and
```
Thanks John, I've included this feedback.

ogradyp commented on modules/eco-setting-node-maintenance-cr-cli.adoc:
```diff
@@ -38,14 +38,16 @@ spec:
 ----
 +
-.Example output
+. Check the example output
```
Thanks John, I've included this feedback.

razo7 commented:
I am having second doubts about the current layout. After talking together I believe we should move from this layout:

```
1. About the Node Maintenance Operator
2. Maintaining bare-metal nodes
3. Installing the Node Maintenance Operator
    3a. Installing the Node Maintenance Operator by using the web console
    3b. Installing the Node Maintenance Operator by using the CLI
4. Setting a node to maintenance mode
    4a. Setting a node to maintenance mode by using the web console
    4b. Setting a node to maintenance mode by using the CLI
    4c. Setting a node to maintenance mode by using the web console using the Actions Control
    4d. Checking status of current NodeMaintenance CR tasks
5. Resuming a node from maintenance mode
    5a. Resuming a node from maintenance mode by using the web console
    5b. Resuming a node from maintenance mode by using the CLI
    5c. Resuming a node from maintenance mode by using the web console using the Actions Control
6. Additional resources
```

To this layout:
```
1.  About the Node Maintenance Operator
2.  Installing the Node Maintenance Operator
     2a. Installing the Node Maintenance Operator by using the web console
     2b. Installing the Node Maintenance Operator by using the CLI
3. Setting a node to maintenance mode
     3a. Setting a node to maintenance mode by using the web console
     3b. Setting a node to maintenance mode by using the CLI
     3c. Checking status of current NodeMaintenance CR tasks
4. Resuming a node from maintenance mode
     4a. Resuming a node from maintenance mode by using the web console
     4b. Resuming a node from maintenance mode by using the CLI
5. Working with bare-metal nodes
     5a. Maintaining bare-metal nodes
     5b. Setting a node to maintenance mode
     5c. Resuming a node from maintenance mode
6. Additional resources
```

Can you apply these changes? @ogradyp

razo7 commented:
I would like to delete the second sentence in the beginning of the page - http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html
`
This is a standalone version of the Node Maintenance Operator that is independent of OpenShift Virtualization installation.
`

ogradyp commented:
> I would like to delete the second sentence in the beginning of the page - http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html `This is a standalone version of the Node Maintenance Operator that is independent of OpenShift Virtualization installation.`

@razo7 This is now updated, see http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html

ogradyp commented:
> I am having second doubts about the current layout. After talking together I believe we should move from this layout:
> 
> ```
> 1. About the Node Maintenance Operator
> 2. Maintaining bare-metal nodes
> 3. Installing the Node Maintenance Operator
>     3a. Installing the Node Maintenance Operator by using the web console
>     3b. Installing the Node Maintenance Operator by using the CLI
> 4. Setting a node to maintenance mode
>     4a. Setting a node to maintenance mode by using the web console
>     4b. Setting a node to maintenance mode by using the CLI
>     4c. Setting a node to maintenance mode by using the web console using the Actions Control
>     4d. Checking status of current NodeMaintenance CR tasks
> 5. Resuming a node from maintenance mode
>     5a. Resuming a node from maintenance mode by using the web console
>     5b. Resuming a node from maintenance mode by using the CLI
>     5c. Resuming a node from maintenance mode by using the web console using the Actions Control
> 6. Additional resources
> ```
> 
> To this layout:
> 
> ```
> 1.  About the Node Maintenance Operator
> 2.  Installing the Node Maintenance Operator
>      2a. Installing the Node Maintenance Operator by using the web console
>      2b. Installing the Node Maintenance Operator by using the CLI
> 3. Setting a node to maintenance mode
>      3a. Setting a node to maintenance mode by using the web console
>      3b. Setting a node to maintenance mode by using the CLI
>      3c. Checking status of current NodeMaintenance CR tasks
> 4. Resuming a node from maintenance mode
>      4a. Resuming a node from maintenance mode by using the web console
>      4b. Resuming a node from maintenance mode by using the CLI
> 5. Working with bare-metal nodes
>      5a. Maintaining bare-metal nodes
>      5b. Setting a node to maintenance mode
>      5c. Resuming a node from maintenance mode
> 6. Additional resources
> ```
> 
> Can you apply these changes? @ogradyp

@razo7 These changes are now applied, see http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#working-with-bare-metal-nodes

razo7 commented on nodes/nodes/eco-node-maintenance-operator.adoc:
```diff
@@ -6,17 +6,15 @@ include::_attributes/common-attributes.adoc[]
 toc::[]
 
-You can use the Node Maintenance Operator to place nodes in maintenance mode. This is a standalone version of the Node Maintenance Operator that is independent of {VirtProductName} installation.
+You can use the Node Maintenance Operator to place nodes in maintenance mode.
```
> > I would like to delete the second sentence in the beginning of the page - http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html `This is a standalone version of the Node Maintenance Operator that is independent of OpenShift Virtualization installation.`
> 
> @razo7 This is now updated, see http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html

Now the `You can use the Node Maintenance Operator to place nodes in maintenance mode.` sentence doesn't add anything, that we don't say under `About the Node Maintenance Operator`.
I think it would be better to delete the whole sentence and the about section, `About the Node Maintenance Operator`,  and let the page starts with `You can place nodes into maintenance mode using the oc adm utility, or using NodeMaintenance custom resources (CRs).`

ogradyp replied:
Updated as requested http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html

razo7 commented on nodes/nodes/eco-node-maintenance-operator.adoc:
```diff
@@ -6,17 +6,15 @@ include::_attributes/common-attributes.adoc[]
 [NOTE]
 ====
-If you have installed {VirtProductName}, you must use the Node Maintenance Operator that is bundled with it.
+If {VirtProductName} v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version. You must first upgrade {VirtProductName}, and then install the latest version of the Node Maintenance Operator from the `OperatorHub`.
```
I would move the NOTE from the beginning to be under [Installing the Node Maintenance Operator](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-standalone-nmo) since it is about installing.

razo7 replied:
By looking in https://issues.redhat.com/browse/CNV-17623?focusedCommentId=20673382&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-20673382 I see that CNV suggest that a user can and should use "our" NMO before doing an upgrade if he/she want the node to stay under maintenance while doing the CNV upgrade...

I would suggest to leave the note, without an action from the user, as follows:


> If OpenShift Virtualization v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version.

ogradyp replied:
Updated as requested http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-standalone-nmo

razo7 commented on modules/eco-setting-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,28 @@
+= Setting a bare-metal node to maintenance mode
+Set a bare-metal node to maintenance mode using the Options menu {kebab} found on each node in the *Compute* -> *Nodes* list, or using the *Actions* control of the *Node Details* screen.
+
+[NOTE]
```
This NOTE is no longer needed, since this NOTE is already under `Working with bare-metal nodes`. Please delete this NOTE.

ogradyp replied:
Removed as requested http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-actions-web-console_node-maintenance-operator

razo7 commented on nodes/nodes/eco-node-maintenance-operator.adoc:
```diff
@@ -50,9 +47,19 @@ include::modules/eco-resuming-node-maintenance-cr-cli.adoc[leveloffset=+2]
 
+[id="working-with-bare-metal-nodes"]
+== Working with bare-metal nodes
+In addition to maintaining virtual nodes, the Node Maintenance Operator can also maintain bare-metal nodes. For clusters with bare-metal nodes, you can place a node into maintenance mode, and resume a node from maintenance mode by using the web console Actions control.
```
For clusters with bare-metal nodes, you can place a node into maintenance mode, and resume a node from maintenance mode by using the web console Actions control. These methods are additional to the above methods and only valid for bare-metal cluster.

razo7 replied:
Then add a NOTE

> These methods, setting and resuming `NodeMaintenance` CR, are additional to the above methods and only valid for bare-metal cluster.

razo7 replied:
@ogradyp @anna-savina WDYT?
I think that without this note the user might think that only the methods under `Working with bare-metal nodes` are valid for bare-metal nodes/cluster.

ogradyp replied:
Updated as requested http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#working-with-bare-metal-nodes

razo7 commented:
LGTM

razo7 commented:
Very nice work @ogradyp !

jeana-redhat commented on modules/eco-resuming-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,24 @@
+
+.Verification
+
+Navigate to the *Compute* -> *Nodes* page and verify that the corresponding node has a status of `Ready`.
```
```suggestion
* Navigate to the *Compute* -> *Nodes* page and verify that the corresponding node has a status of `Ready`.
```

ogradyp replied:
Thanks @jeana-redhat , updated as suggested

jeana-redhat commented on modules/eco-resuming-node-maintenance-cr-cli.adoc:
```diff
@@ -28,3 +28,22 @@ $ oc delete -f nodemaintenance-cr.yaml
+
+.Verification
+
+. Check the progress of the maintenance task by running the following command, replacing `<node-name>` with the name of your node; for example, `node-1.example.com`:
++
+[source,terminal]
+----
+$ oc describe node node-1.example.com
+----
++
```
```suggestion
. Check the progress of the maintenance task by running the following command:
+
[source,terminal]
----
$ oc describe node <node-name>
----
+
where `<node-name>` is the name of your node.
```

ogradyp replied:
Thanks @jeana-redhat , updated as suggested

jeana-redhat commented on modules/eco-resuming-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,24 @@
+.Procedure
+
+. From the *Administrator* perspective, in the web console, click *Compute* -> *Nodes*.
+. You can resume the node from this screen, which makes it easier to perform actions on multiple nodes, or from the *Node Details* screen where you can view comprehensive details of the selected node:
```
```suggestion
. You can resume the node from this screen, which makes it easier to perform actions on multiple nodes, or from the *Node Details* screen, where you can view comprehensive details of the selected node:
```

ogradyp replied:
Thanks @jeana-redhat , updated as suggested

jeana-redhat commented on nodes/nodes/eco-node-maintenance-operator.adoc:
```diff
@@ -6,21 +6,19 @@ include::_attributes/common-attributes.adoc[]
-====
-If you have installed {VirtProductName}, you must use the Node Maintenance Operator that is bundled with it.
-====
+You can use the Node Maintenance Operator to place nodes in maintenance mode, by using the `oc adm` utility, or `NodeMaintenance` custom resources (CRs).
```
```suggestion
You can use the Node Maintenance Operator to place nodes in maintenance mode by using the `oc adm` utility or `NodeMaintenance` custom resources (CRs).
```

ogradyp replied:
Thanks @jeana-redhat , updated as suggested

jeana-redhat commented on nodes/nodes/eco-node-maintenance-operator.adoc:
```diff
@@ -6,21 +6,19 @@ include::_attributes/common-attributes.adoc[]
 
+[NOTE]
+====
+If {VirtProductName} v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version.
```
Do they need to do anything about it in this case?

```suggestion
If {VirtProductName} version 4.10 or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator.
```

ogradyp replied:
Thanks @jeana-redhat , updated as suggested

ogradyp replied:
No, no action required. 

Feedback from SME earlier:
By looking in https://issues.redhat.com/browse/CNV-17623?focusedCommentId=20673382&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-20673382 I see that CNV suggest that a user can and should use "our" NMO before doing an upgrade if he/she want the node to stay under maintenance while doing the CNV upgrade...

I would suggest to leave the note, without an action from the user, as follows:

If OpenShift Virtualization v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version.

jeana-redhat commented on nodes/nodes/eco-node-maintenance-operator.adoc:
```diff
@@ -29,23 +27,37 @@ The Node Maintenance Operator is supported in a restricted network environment.
 
+[id="working-with-bare-metal-nodes"]
+== Working with bare-metal nodes
+For clusters with bare-metal nodes, you can place a node into maintenance mode, and resume a node from maintenance mode, by using the web console Actions control.
```
```suggestion
For clusters with bare-metal nodes, you can place a node into maintenance mode, and resume a node from maintenance mode, by using the web console *Actions* control.
```

ogradyp replied:
Thanks @jeana-redhat , updated as suggested

jeana-redhat commented on nodes/nodes/eco-node-maintenance-operator.adoc:
```diff
@@ -29,23 +27,37 @@ The Node Maintenance Operator is supported in a restricted network environment.
+
+[NOTE]
+====
+Clusters with bare-metal nodes can also place a node into maintenance mode, and resume a node from maintenance mode, by using the web console and CLI, as outlined. These methods, by using the web console Actions control, are applicable to bare-metal clusters only.
```
```suggestion
Clusters with bare-metal nodes can also place a node into maintenance mode, and resume a node from maintenance mode, by using the web console and CLI, as outlined. These methods, by using the web console *Actions* control, are applicable to bare-metal clusters only.
```

ogradyp replied:
Thanks @jeana-redhat , updated as suggested

jeana-redhat reviewed (commented):
A few stray formatting things, overall looking good :+1:

ogradyp commented on modules/eco-resuming-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,24 @@
+
+.Procedure
+
+. From the *Administrator* perspective, in the web console, click *Compute* -> *Nodes*.
```
Thanks @jeana-redhat , updated as suggested

ogradyp commented on modules/eco-setting-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,23 @@
+
+.Procedure
+
+. From the *Administrator* perspective, in the web console, click *Compute* -> *Nodes*.
```
Thanks @jeana-redhat , updated as suggested

ogradyp commented on modules/eco-setting-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,23 @@
+.Procedure
+
+. From the *Administrator* perspective, in the web console, click *Compute* -> *Nodes*.
+. You can set the node to maintenance from this screen, which makes it easier to perform actions on multiple nodes, or from the *Node Details* screen where you can view comprehensive details of the selected node:
```
Thanks @jeana-redhat , updated as suggested

ogradyp commented on modules/eco-setting-node-maintenance-actions-web-console.adoc:
```diff
@@ -0,0 +1,23 @@
+
+.Verification
+
+Navigate to the *Compute* -> *Nodes* page and verify that the corresponding node has a status of `Under maintenance`.
```
Thanks @jeana-redhat , updated as suggested

ogradyp commented on modules/eco-setting-node-maintenance-cr-cli.adoc:
```diff
@@ -38,14 +38,16 @@ spec:
 ----
 $ oc describe node node-1.example.com
 ----
 +
```
Thanks @jeana-redhat , updated as suggested

ogradyp commented on modules/eco-resuming-node-maintenance-cr-cli.adoc:
```diff
@@ -28,3 +28,22 @@ $ oc delete -f nodemaintenance-cr.yaml
+Events:
+  Type     Reason                  Age                   From     Message
+  ----     ------                  ----                  ----     -------
+  Normal   NodeSchedulable         2m                    kubelet  Node node-1.example.com status is now: NodeSchedulable
```
Ok, thanks

jeana-redhat commented:
/cherrypick enterprise-4.11
